js/packages/common/src/contexts/connection.tsx

```diff
@@ -282,6 +290,8 @@ export const sendTransactions = async (
   failCallback: (reason: string, ind: number) => boolean = (txid, ind) => false,
   block?: BlockhashAndFeeCalculator,
 ): Promise<number> => {
+  if (!wallet.publicKey) throw new WalletNotConnectedError();
```

Comment:
These type guards are because `wallet.publicKey` may be null. In practice, this condition likely won't be reached, because if the underlying wallet disconnects, the adapter and context provider will detect this.